deepcell_tracking/tracking.py

```diff
@@ -99,320 +125,487 @@ def __init__(self,
+        """
+        new_track = len(self.tracks)
+        new_label = new_track + 1
+        self.tracks[new_track] = {
+            'label': new_label,
+            'frames': [frame],
+            'frame_labels': [old_label],
+            'daughters': [],
+            'capped': False,
+            'frame_div': None,
+            'parent': None,
+        }
 
-        if frame > 0 and np.any(self.y[frame] == new_label):
+        if frame > 0 and np.any(self._get_frame(self.y, frame) == new_label):
             raise Exception('new_label already in annotated frame and frame > 0')
 
-        self.y[frame][self.y[frame] == old_label] = new_label
+        if self.data_format == 'channels_first':
+            self.y[:, frame][self.y[:, frame] == old_label] = new_label
+        else:
+            self.y[frame][self.y[frame] == old_label] = new_label
 
     def _initialize_tracks(self):
         """Intialize the tracks. Tracks are stored in a dictionary.
         """
-        self.tracks = {}
-        unique_cells = np.unique(self.y[0])
+        frame = 0  # initial frame
+        unique_cells = self.get_cells_in_frame(frame)
+        self.frame_features = self.get_frame_features(frame, unique_cells)
+        for cell_idx, cell_label in enumerate(unique_cells):
+            self._create_new_track(frame, cell_label)
+            track_id = max(self.tracks)  # newly added track
+            for f in self.frame_features:
+                self.tracks[track_id][f] = self.frame_features[f][[cell_idx]]
 
-        # Remove background that has value 0
-        unique_cells = np.delete(unique_cells, np.where(unique_cells == 0))
+        # Start a tracked label array
+        self.y_tracked = self.y[[frame]].astype('int32')
 
-        for track_counter, label in enumerate(unique_cells):
-            self._create_new_track(0, label)
+    def compute_distance(self, track_centroids, frame_centroids):
+        """Computes the distance between two centroids.
 
-        # Start a tracked label array
-        self.y_tracked = self.y[[0]].astype('int32')
+        Args:
+            track_centroids (tuple): x and y centroid for the given track.
+            frame_centroids (tuple): x and y centroid for the given frame.
 
-    def _compute_feature(self, feature_name, track_feature, frame_feature):
+        Returns:
+            tuple: the distances for tracks and frames,
+                and a boolean indicating whether the distance is valid.
         """
-        Given a track and frame feature, compute the resulting track and frame features.
-        This also returns True or False as the third element of the tuple indicating if these
-        features should be used at all. False indicates that this pair of track & cell features
-        should result in a maximum cost assignment.
-        This is usually for some preprocessing in case it is desired. For example, the
-        distance feature normalizes distances.
+        centroids = np.concatenate([
```

Comment:
It may be worth investigating the way we calculate this distance and the need for concatenate

Reply:
Added an issue to track this, #13.